api/provider/aws/common.go

```diff
@@ -9,6 +9,21 @@ import (
 	"github.com/quintilesims/layer0/api/tag"
 )
 
+func lookupEntityEnvironmentID(store tag.Store, entityType, entityID string) (string, error) {
+	tags, err := store.SelectByTypeAndID(entityType, entityID)
+	if err != nil {
+		return "", err
+	}
+
+	tag, ok := tags.WithKey("environment_id").First()
+	if !ok {
+		// is this canonical error handling?
+		return "", fmt.Errorf("Cannot resolve environment_id for %s '%s'", entityType, entityID)
```

Comment:
We use `common/errors` when we want to capture meaningful error codes.

Speaking of,  this will likely be one of the first helper functions called for many of the `Read()` functions. 
If the entity does not exist, we won't capture that error code correctly. That will break the terraform provider since it checks for specific `errors.<Entity>DoesNotExist` error codes.

Perhaps we can check if `len(tags) == 0`? We would probably need a new helper function in `common/errors` that can do a  `switch` on `entityType string` and convert it to the proper error code.

Reply:
I had blanked on the pattern before, but I think I was looking for something more like this:
```
    tag, ok := tags.WithKey("environment_id").First(); ok {
        return tag.Value, nil
    }

    // handle error here
```

Reply:
@zpatrick Do we need to check `len(tags) == 0`? Is `!ok` not enough to result in a DNE error?

I've updated to `return "", errors.Newf(errors.EnvironmentDoesNotExist, "Could not resolve environment ID for %s '%s'", entityType, entityID)` on `!ok`, but maybe that's wrong. Should we be reporting that the entity doesn't exist instead of the environment?